src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+    }
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::prepare_for_new_conversation() {
+    // FIXME: for input_ids it must be padding from tokenizer that not available from here
+    // Get it from NPUW options
+    LOG_DEBUG("ov::npuw::LLMInferRequest::prepare_for_new_conversation()");
+    LOG_BLOCK();
+
+    auto prefill_compiled = m_prefill_request->get_compiled_model();
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("input_ids")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("attention_mask")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("position_ids")), 0u);
+    fill_tensor<int64_t>(m_kvcache_request->get_tensor(m_kvcache_in_ports.at("attention_mask")), 0u);
+    m_kvcache_desc.num_stored_tokens = 0u;
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::infer_prefill(ov::SoPtr<ov::ITensor> input_ids,
+                                              ov::SoPtr<ov::ITensor> attention_mask,
+                                              ov::SoPtr<ov::ITensor> position_ids) {
+    LOG_DEBUG("LLMInferRequest::infer_prefill()");
+    LOG_BLOCK();
+
+    prepare_for_new_conversation();
+
+    auto prefill_compiled = m_prefill_request->get_compiled_model();
+
+    auto padded_input_ids = m_prefill_request->get_tensor(m_prefill_in_ports.at("input_ids"));
+    const size_t offset = padded_input_ids->get_size() - input_ids->get_size();
+    std::copy_n(input_ids->data<int64_t>(),
+                input_ids->get_size(),
+                padded_input_ids->data<int64_t>() + offset);
+
+    auto padded_attention_mask = m_prefill_request->get_tensor(m_prefill_in_ports.at("attention_mask"));
+    std::copy_n(attention_mask->data<int64_t>(),
+                attention_mask->get_size(),
+                padded_attention_mask->data<int64_t>() + offset);
+
+    auto padded_position_ids = m_prefill_request->get_tensor(m_prefill_in_ports.at("position_ids"));
+    std::copy_n(position_ids->data<int64_t>(),
+                position_ids->get_size(),
+                padded_position_ids->data<int64_t>() + offset);
+
+    m_prefill_request->infer();
+    m_kvcache_desc.num_stored_tokens += static_cast<uint32_t>(input_ids->get_size());
+    m_need_copy_kvcache = true;
+
+    m_logits = m_prefill_request->get_tensor(m_prefill_out_ports.at("logits"));
+
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::infer_generate(ov::SoPtr<ov::ITensor> input_ids,
+                                               ov::SoPtr<ov::ITensor> attention_mask,
+                                               ov::SoPtr<ov::ITensor> position_ids) {
+    LOG_DEBUG("LLMInferRequest::infer_generate()");
+    LOG_BLOCK();
+
+    // NB: KV-cache is full, further generation is impossible
+    if (m_kvcache_desc.num_stored_tokens == m_kvcache_desc.total_size) {
+        LOG_DEBUG("KV-Cache is full");
```

Comment:
This should be some common log I believe, if not an exception we could raise.

Reply:
Added `OPENVINO_THROW` here.